tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/tests/PitclipsePitSummaryViewTest.java

```diff
@@ -0,0 +1,75 @@
+package org.pitest.pitclipse.ui.tests;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.equalTo;
+import static org.pitest.pitclipse.ui.behaviours.pageobjects.PageObjects.PAGES;
+import static org.pitest.pitclipse.ui.view.PitView.BLANK_PAGE;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.swtbot.swt.finder.junit.SWTBotJunit4ClassRunner;
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pitest.pitclipse.ui.behaviours.pageobjects.PitSummaryView;
+import org.pitest.pitclipse.ui.view.PitView;
+
+@RunWith(SWTBotJunit4ClassRunner.class)
+public class PitclipsePitSummaryViewTest extends AbstractPitclipseSWTBotTest {
+    private static final String TEST_PROJECT = "org.pitest.pitclipse.testprojects.twoclasses";
+    private static final String FOO_BAR_PACKAGE = "foo.bar";
+    private static final String FOO_BAR_PACKAGE_RESULT = "foo.bar";
+    private static final String FOO_CLASS_HEADER = "Foo.java";
+    private static final String FOO_CLASS_RESULT_URL_END = FOO_CLASS_HEADER + ".html";
+    private static final String INDEX_URL_END = "index.html";
+    private static final String INDEX_HEADER = "Pit Test Coverage Report";
+    private PitSummaryView summaryView;
+
+    @BeforeClass
+    public static void setupJavaProject() throws CoreException {
+        importTestProject(TEST_PROJECT);
+    }
+
+    @Before
+    public void resetView() throws InterruptedException {
+        // If this class is run alone, the view can't be closed and a RuntimeException
+        // is thrown and expected
+        closeViewById(PitView.VIEW_ID);
+        openViewById(PitView.VIEW_ID);
+        summaryView = PAGES.getPitSummaryView();
+    }
+
+    @Test
+    public void navigateWithButtons() throws CoreException {
+        final long timeout = SWTBotPreferences.TIMEOUT;
+        try {
+            // set timeout to small time, because offline page loads should be quick and
+            // some pages are expected to not change and need timeout
+            SWTBotPreferences.TIMEOUT = 50;
+            // should not change page
+            assertThat(summaryView.clickHome(BLANK_PAGE), equalTo(BLANK_PAGE));
+            assertThat(summaryView.clickForward(BLANK_PAGE), equalTo(BLANK_PAGE));
+            assertThat(summaryView.clickBack(BLANK_PAGE), equalTo(BLANK_PAGE));
```

Comment:
In my testing, `resetView` worked good and it was only added because it was needed for the test running via maven, because the view held the results, as you suspected, even after closing.

Reply:
So if I understand you well you're saying that `resetView` actually clears the results from the view, and that it's not the action of closing the view that clears the results? How are the results cleared, in that case?

Reply:
This is correct.
It is not cleared while closing, it is reset while opening. See: `PitView.createPartControl`

Reply:
OK, so technically it's not the test that clears the view: the view is _always_ cleared when it opens, even outside our tests. That's not the behavior I'd expect but it looks like it's not a regression so don't bother fixing that in this PR.

So since the view is effectively cleared tests shouldn't interfere with each other. Could you put the asserts into separate test methods?

Reply:
Will do

Reply:
> Ok, I missed those new asserts because I was expecting each of them to be into its own test method. It's usually good practice to isolate single test cases because that makes failures easier to identify and tests easier to understand.
> @LorenzoBettini do you have an opinion on these topics?

If I'm still on time answering this, I'm fine with several assertions in the same test method if
- they are related to the same scenario/functionality under test
- they speed up things a bit since the setup is heavy

I would say that this is such a situation: I'd be OK with the 3 asserts in the same test method.

Again, if that has already been changed or you prefer the other way round, forget about what I've just said ;)

Reply:
It is already implemented in a new method.